cli/command/utils.go

```diff
@@ -78,7 +78,10 @@ func PrettyPrint(i any) string {
 
 type PromptError error
 
-var ErrPromptTerminated = PromptError(errors.New("prompt terminated"))
+var (
+	ErrPromptTerminated       = PromptError(errdefs.Cancelled(errors.New("prompt terminated")))
+	ErrUserCancelledOperation = PromptError(errdefs.Cancelled(errors.New("user cancelled operation")))
+)
```

Comment:
Somewhat similar here, I _guess_ the main reason here is for consistency. On its own that's good; we must be careful not to put _too much_ value to it; exporting the error comes with a "load" / "expectation" to it which others may now start to depend on.

This can be risky; for a long time (and there might still be code in place) we ended up with errors that _had to match_ a specific string (`No such` for "not found errors"), which spread across many code-bases, and now meant we could not change errors without breaking code-flows.

So if these errors are _only_ for consistency, and _not_ special / sentinel errors, we could still have a (non exported?) utility to price the error (`return someUtilityToProduceError()`), but prevent that from becoming a _must_ (for code flow).

While we _should_ prefer consistency from a UX perspective; there may be exceptions to the rule (a friendlier, but still recognisable error, which could be specific to the action at hand (_"pruning buildcache is cancelled"_  / _"pruning volumes is cancelled"_).


The `errdefs` ("interface approach") types were created from that perspective; they would allow the flexibility of changing the (user-facing?) error _message_ without breaking business logic (and the "interface" approach allows for custom implementations of the interface, without enforcing a specific implementation / type / definition to be  imported).

Reply:
Hmm I see, in this case I used the exported error message for consistency. I guess we don't need to return the same error message every time a user enters a "no" on the prompt like here
https://github.com/docker/cli/blob/63bf7dec61bbb4d1f9dbbb8133367d6aeae762b5/cli/command/volume/prune.go#L84-L86

Since the prompt will print a new line on an stdin and we take care of printing a new line on termination, we could theoretically just return a nil when the prompt returns false. 

```

            ---- True --- continue code
Prompt -- > ---- False --- return nil
            ---- Terminate --- return defined error

Reply:
ended up returning `errdefs.Cancelled` with a custom error message per command that had the Prompt return `false`. This still allows us to exit with code 0 and atm it won't print the error message. We could opt for printing it if we'd like, but i don't think it's necessary right now.